File: java/amp-rabbit/src/main/java/amp/rabbit/connection/BasicConnectionFactory.java
package amp.rabbit.connection;


import amp.rabbit.topology.Broker;

import com.rabbitmq.client.ConnectionFactory;


public class BasicConnectionFactory extends BaseConnectionFactory {

	protected String username;
	protected String password;
	
	public BasicConnectionFactory(String username, String password) {
		
		super();

		this.username = username;
		this.password = password;
	}
	

	@Override
	public void configureConnectionFactory(ConnectionFactory factory, Broker broker) throws Exception {
    	super.configureConnectionFactory(factory, broker);

        factory.setUsername(username);
        factory.setPassword(password);
	}

}

File: dotnet/source/amp.rabbit/connection/BasicConnectionFactory.cs
using amp.rabbit.topology;
using RabbitMQ.Client;

namespace amp.rabbit.connection
{
    public class BasicConnectionFactory : BaseConnectionFactory
    {
        protected string _username;
        protected string _password;


        public BasicConnectionFactory(string username = "guest", string password = "guest")
        {
            _username = username;
            _password = password;
        }


        public override void ConfigureConnectionFactory(ConnectionFactory factory, Broker broker)
        {
            base.ConfigureConnectionFactory(factory, broker);

            factory.UserName = _username;
            factory.Password = _password;
        }

    }
}
